### PR TITLE
Mepts 10124 tpt completion changes

### PR DESCRIPTION
**Pull request by ItsMurumba**

### CODE REVIEW

**Review by TinoMuianga — COMMENTED** [review]
Added Section B INH Final after merging with TPT ELigible

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java [hunk 1]
```diff
@@ -280,19 +399,75 @@ public DataDefinition getPatientAnd3HPInitiationDateOnFichaClinica() {
...
+            + "       INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "       INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "       INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id "
+            + "       INNER JOIN obs o3 ON e.encounter_id = o3.encounter_id "
+            + "WHERE  p.voided = 0 AND e.voided = 0 AND o.voided = 0 AND o2.voided = 0 AND o3.voided = 0 "
+            + "       AND e.location_id = :location "
+            + "       AND e.encounter_type = ${6} "
+            + "       AND ( o.concept_id = ${1719} AND o.value_coded = ${23954} ) "
+            + "       OR  ( (o2.concept_id = ${23985} AND o2.value_coded = ${23954}) "
+            + "             AND (o3.concept_id = ${165308} AND o3.value_coded = ${1256}) ) "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java]:
```suggestion
            + "       AND (( o.concept_id = ${1719} AND o.value_coded = ${23954} ) "
            + "       OR  ( (o2.concept_id = ${23985} AND o2.value_coded = ${23954}) "
            + "             AND (o3.concept_id = ${165308} AND o3.value_coded = ${1256}) ) )"
```

Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java]: There's a need to put those brackets for it to work correctly

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java [hunk 2]
```diff
@@ -280,19 +399,75 @@ public DataDefinition getPatientAnd3HPInitiationDateOnFichaClinica() {
...
+            + "       Min(o2.value_datetime) AS earliest_date "
+            + "FROM   patient p "
+            + "       INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "       INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "       INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id "
+            + "WHERE  p.voided = 0 AND e.voided = 0 AND o.voided = 0 AND o2.voided = 0 "
+            + "       AND e.location_id = :location "
+            + "       AND e.encounter_type = ${53} "
+            + "       AND o.concept_id = ${23985} AND o.value_coded = ${23954} "
+            + "       AND o2.concept_id = ${6128} AND o2.value_datetime < CURRENT_DATE() "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java]: They said during the reporting period, why using CURRENT_DATE

**Review by psmatsinhe — COMMENTED** [review]
Please have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java (hunk 1 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java:420]: Solved the brackets issue.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java (hunk 2 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationDataDefinitionQueries.java:469]: Replaced the CURRENT_DATE with :startDate AND :endDate

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 3]
```diff
@@ -653,42 +813,41 @@ public CohortDefinition getINHStartA1(
...
             + " p.voided = 0 AND e.voided = 0 AND o.voided = 0"
-            + "    AND e.encounter_type = ${6}"
-            + "    AND o.concept_id = ${6122}"
-            + "    AND o.value_coded = ${1256}"
-            + "    AND e.encounter_datetime < :endDate"
-            + "    AND e.location_id = :location";
+            + " AND e.encounter_type = ${53}"
+            + " AND o.concept_id = ${23985}"
+            + " AND o.value_coded = ${23954}"
+            + " AND e.encounter_datetime < :endDate"
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java]: @ItsMurumba In this form we don't use encounter_datetime, usually we go with obs_datetime

Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java]:
```suggestion
            + " AND o.obs_datetime < :endDate"
```

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 4]
```diff
@@ -701,32 +860,41 @@ public CohortDefinition getINHStartA2(
...
+            + "     INNER JOIN"
+            + "  encounter e ON p.patient_id = e.patient_id"
+            + "     INNER JOIN"
+            + "  obs o ON e.encounter_id = o.encounter_id"
+            + " WHERE"
+            + " p.voided = 0 AND e.voided = 0 AND o.voided = 0"
+            + " AND e.encounter_type = ${53}"
+            + " AND o.concept_id = ${23985}"
+            + " AND o.value_coded = ${656}"
+            + " AND o.value_datetime < :endDate"
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java]: @ItsMurumba I dont think this is possible, for one obs you have a concept and one answer, a value coded or value datetime not both.
That being said this query will return empty

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 5]
```diff
@@ -701,32 +860,41 @@ public CohortDefinition getINHStartA2(
...
-   *   <li>A3: Select all patients with Ficha clinica (encounter type 6) with “Profilaxia com INH”
-   *       (concept id 6128) value datetime before end date
+   *   <li>A -A1.2: Select all patients with Ficha Resumo (encounter type 53) with Última
+   *       profilaxia(concept id 23985) value coded INH(concept id 656)
    *   <li>
    *
    * @return CohortDefinition
    */
-  public CohortDefinition getINHStartA3(int encounterType, int profilaxiaIsoniazidaConcept) {
+  public CohortDefinition getINHStartA1Part2A(
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:869]: @ItsMurumba why did you remove the previous method? don't we need it elsewhere? I think the most correct here was to create a different one not to remove the previous, reason being that they doing totally different things

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 6]
```diff
@@ -739,30 +907,172 @@ public CohortDefinition getINHStartA3(int encounterType, int profilaxiaIsoniazid
...
+   *       (concept id 6122) with value code “Inicio” (concept id 1256) and encounter datetime
+   *       before end date
    *   <li>
    *
    * @return CohortDefinition
    */
-  public CohortDefinition getINHStartA4(
-      int pediatriaSeguimentoEncounterType, int dataInicioProfilaxiaIsoniazidaConcept) {
-
+  public CohortDefinition getINHStartA2Part1(
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:916]: Same here, no need to remove

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 7]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+  }
+
+  /**
+   * <b>IMER1</b>: User_Story_ TPT <br>
+   *
+   * <ul>
+   *   <li>C7 Part 1: Select all patients with “Regime de TPT” (concept id 23985) with value coded
+   *       “3HP” or ” 3HP+Piridoxina” (concept id in [23954, 23984]) and “Seguimento de tratamento
+   *       TPT”(concept ID 23987) value coded “continua” or “fim” or no value(concept ID in [1257,
+   *       1267, null]) marked on the first FILT (encounter type 60) and encounter datetime before
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:1787]: @ItsMurumba I dont see this section of first in the code. MIN ()

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 8]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+            + "AND o.concept_id = ${23985} AND o.value_coded IN(${23954}, ${23984})   "
+            + "AND e.location_id = :location       "
+            + "AND e.encounter_datetime < :endDate "
+            + "AND p.voided = 0 AND e.voided = 0 AND o.voided = 0   "
+            + ") AS regimeTPT    "
+            + "ON regimeTPT.patient_id = p.patient_id   "
+            + "WHERE e.encounter_type = ${60}   "
+            + "AND o.concept_id = ${23985} AND o.value_coded IN(${23954},${23984})   "
+            + "AND e.location_id = :location   "
+            + "AND e.encounter_datetime BETWEEN DATE_SUB(regimeTPT.encounter_datetime, INTERVAL 4 MONTH) AND regimeTPT.encounter_datetime   "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:1909]: @ItsMurumba please lets confirm with the BA if this is what they mean

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 9]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+            + "AND e.encounter_datetime < :endDate "
+            + "AND p.voided = 0 AND e.voided = 0 AND o.voided = 0   "
+            + ") AS regimeTPT    "
+            + "ON regimeTPT.patient_id = p.patient_id   "
+            + "WHERE e.encounter_type = ${6}   "
+            + "AND (o.concept_id = ${23985} AND o.value_coded = ${23954}) "
+            + "AND (o2.concept_id = ${165308} AND o2.value_coded = ${1256})   "
+            + "OR (o.concept_id = ${1719} AND o.value_coded IN (${23954},${165307})) "
+            + "AND e.location_id = :location   "
+            + "AND e.encounter_datetime BETWEEN DATE_SUB(regimeTPT.encounter_datetime, INTERVAL 4 MONTH) AND regimeTPT.encounter_datetime   "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:1980]: Would be great if we discuss this

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 10]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+            + "AND o.concept_id = ${23985} AND o.value_coded = ${23954} "
+            + "AND e.location_id = :location       "
+            + "AND e.encounter_datetime < :endDate "
+            + "AND p.voided = 0 AND e.voided = 0 AND o.voided = 0   "
+            + ") AS regimeTPT    "
+            + "ON regimeTPT.patient_id = p.patient_id   "
+            + "WHERE e.encounter_type = ${53}   "
+            + "AND o.concept_id = ${23985} AND o.value_coded = ${23954} "
+            + "AND e.location_id = :location   "
+            + "AND o.value_datetime BETWEEN DATE_SUB(regimeTPT.encounter_datetime, INTERVAL 4 MONTH) AND regimeTPT.encounter_datetime   "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:2037]: Same

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 11]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+
+    String query =
+        " SELECT p.patient_id FROM patient p    "
+            + "INNER JOIN encounter e ON p.patient_id  = e.patient_id    "
+            + "INNER JOIN obs o ON e.encounter_id = o.encounter_id    "
+            + "INNER JOIN (   "
+            + " SELECT p.patient_id, e.encounter_datetime FROM patient p    "
+            + "INNER JOIN encounter e ON p.patient_id  = e.patient_id    "
+            + "INNER JOIN obs o ON e.encounter_id = o.encounter_id    "
+            + "WHERE e.encounter_type = ${60}    "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:2102]: The encounter 60 and 6 are not in the store and seems like this query is implementing other user store! can you please confirm the user store?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 12]
```diff
@@ -1245,79 +1556,791 @@ public CohortDefinition get3HPStartC3(
...
+            + "             INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id     "
+            + "             WHERE "
+            + "             p.voided = 0 AND e.voided = 0 AND o.voided = 0 "
+            + "                AND e.encounter_type = ${6} "
+            + "             AND (o.concept_id = ${23985} AND o.value_coded = ${23954})    "
+            + "             AND (o2.concept_id = ${165308} AND o2.value_coded = ${1256})    "
+            + "                AND e.encounter_datetime < :endDate "
+            + "                AND e.location_id = :location ) AS regimeTPT    "
+            + "ON regimeTPT.patient_id = p.patient_id   "
+            + "WHERE e.encounter_type = ${53}   "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java]: I dont get this one here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 13]
```diff
@@ -444,9 +480,12 @@ public CohortDefinition getINHStartA1(
     Map<String, Integer> map = new HashMap<>();
     map.put("53", masterCardEncounterType);
     map.put("6128", dataInicioProfilaxiaIsoniazidaConcept);
+    map.put("23985", regimenTpt);
+    map.put("23954", inhRpt);
 
     String query =
-        " SELECT p.patient_id"
+        "SELECT patient_id FROM( "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please update spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 14]
```diff
@@ -481,7 +550,12 @@ public CohortDefinition getINHStartA1(
    * @return CohortDefinition
    */
   public CohortDefinition getINHStartA2(
-      int adultoSeguimentoEncounterType, int startDrugsConcept, int isoniazidUsageConcept) {
+      int adultoSeguimentoEncounterType,
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please update spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 15]
```diff
@@ -607,6 +694,7 @@ public CohortDefinition getINHStartA4(
     map.put("53", masterCardEncounterType);
     map.put("6122", isoniazidUsageConcept);
     map.put("1256", startDrugsConcept);
+    map.put("165308", proflaxisStatus);
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please update spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 16]
```diff
@@ -1028,209 +1131,929 @@ public CohortDefinition getIPTB1(
...
-            + "                                    AND ee.location_id = :location"
-            + "                                    AND ee.encounter_datetime <= :endDate"
-            + "                            LIMIT 1)"
-            + "                        AS DATE),"
-            + "                    INTERVAL 365 DAY)"
-            + "         GROUP BY p.patient_id";
+        "SELECT "
+            + "                            p.patient_id "
+            + "                        FROM "
+            + "                            patient p "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please let's improve this query, ping me when ready

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 17]
```diff
@@ -1028,209 +1131,929 @@ public CohortDefinition getIPTB1(
...
+    map.put("23982", tbMetadata.getIsoniazidePiridoxinaConcept().getConceptId());
+    map.put("23987", hivMetadata.getPatientTreatmentFollowUp().getConceptId());
+    map.put("1705", hivMetadata.getRestartConcept().getConceptId());
+    map.put("1257", hivMetadata.getContinueRegimenConcept().getConceptId());
+
+    String query =
+        "SELECT pp.patient_id FROM patient pp INNER JOIN encounter e ON pp.patient_id = e.patient_id "
+            + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "                                     INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id "
+            + "WHERE pp.voided =0 AND e.voided = 0 AND o.voided=0 AND o2.voided=0 "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Same here, let's improve this query

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@helderjosue and @TinoMuianga here is the review

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 5 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:869]: This method was not removed, it was just moved to line 1019.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 6 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:916]: This method was not removed, it was just moved to line 1058.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 7 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:1787]: Now is implemented

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 8 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:1909]: Seems like thats what they mean, once **FILT 3HP Start Date**  is defined on section C3 from the following  [US](https://docs.google.com/document/d/1wRxiqgcPPUXoDoRr7cG9EKKSeTEnJh3c34Oc6Vxz8FM/edit)

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 11 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:2102]: Those encounters types are from C3! So the specs of C3 are missing in this implementation.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 12 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java]: The same, specs for C5 is missing, and should use encounter_datetime instead of value_datetime and ficha clinica

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 13 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: updated on TPT Eligible ticket.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 14 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: updated on TPT Eligible ticket.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 15 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: updated on TPT Eligible ticket.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 16 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: This query was reimplemented on TPT Eligible ticket.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 17 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: This query was reimplemented on TPT Eligible ticket.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 15 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please put it here @helderjosue

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 14 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please put it here @helderjosue

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 13 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Please put it here @helderjosue

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 16 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: Let's look at it again

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 18]
```diff
@@ -155,7 +134,7 @@ public CohortDefinition getTxCurrWithTPTCompletion() {
             mapping));
 
     compositionCohortDefinition.addSearch(
-        "B1",
+        "B1A",
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:137]: @helderjosue where are you using this?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 19]
```diff
@@ -170,7 +149,10 @@ public CohortDefinition getTxCurrWithTPTCompletion() {
                 tbMetadata.getRegimeTPTConcept().getConceptId(),
                 tbMetadata.getIsoniazidConcept().getConceptId(),
                 tbMetadata.getIsoniazidePiridoxinaConcept().getConceptId()),
-            mapping));
+            mapping3));
+
+    compositionCohortDefinition.addSearch(
+        "B1B", EptsReportUtils.map(tptEligiblePatientListCohortQueries.getIPTB1part2(), mapping3));
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:155]: @helderjosue where are you using this?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 11 above)
Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:2102]: Please include it then

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@helderjosue please have a look on my 3 comments

**Comment by psmatsinhe** [comment]
@helderjosue Do we need this still?

**Comment by helderjosue** [comment]
@psmatsinhe No, those changes were merged on another branch.
